### DISCUSSION TITLE
Containerfile: how to prevent missing build argument warning from spamming the console?

### DISCUSSION THREAD

**Discussion by GrabbenD** [containers/podman#19603] [Q&A] [answered]
# Question
I have multiple static variables which are repeating a lot between stages and I'd like to declare them in one place to make the code easier to follow.

# What I've tried
My current approach to copying variables between stages is `ARG` as seen in this example:
```Dockerfile
ARG BUILDDIR=/mnt

FROM archlinux:base AS rootfs
ARG BUILDDIR
RUN echo Result: $BUILDDIR
# Result: /mnt

FROM scratch AS host
ARG BUILDDIR
COPY --from=rootfs / /
RUN echo Result: ${BUILDDIR}
# Result: /mnt
```

However, this spams a lot of warning messages about missing build args:
> WARN[0000] missing "BUILDDIR" build argument. Try adding "--build-arg BUILDDIR=<VALUE>" to the command line
> WARN[0000] missing "BUILDDIR" build argument. Try adding "--build-arg BUILDDIR=<VALUE>" to the command line

I've also tried `ENV` but the value isn't copied between stages:
```Dockerfile
FROM archlinux:base AS rootfs
ENV BUILDDIR=/mnt
RUN echo Result: $BUILDDIR
# Result: /mnt

FROM scratch AS host
COPY --from=rootfs / /
RUN echo Result: ${BUILDDIR}
# Result: 
```

I know it's just a warning but I'd like to keep messages which needs attention to minimal (warnings/errors/..)

**Is there any other way of copying variables between stages without producing warning messages?**

**Comment by Luap99** [comment]
@flouthoc PTAL

**Comment by flouthoc** [comment, marked as answer]
Thanks this is fixed with: https://github.com/containers/buildah/pull/4983, once recent buildah is vendored you will not see this warnings on globally decalared args.
